### ISSUE TITLE
How to Identify Top 10 Most Abundant Fungi Genera and list them with Relative Abundance.

### ISSUE THREAD

**Issue by sarakn97** [joey711/phyloseq#1614]
Hello,
I need to identify the top 10 most abundant Genus for the Asthma Samples and the Control Samples that I have.

I have the following phyloseq object:

phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 546 taxa and 37 samples ]
sample_data() Sample Data:       [ 37 samples by 12 sample variables ]
tax_table()   Taxonomy Table:    [ 546 taxa by 7 taxonomic ranks ]
refseq()      DNAStringSet:      [ 546 reference sequences ]

14 of the Samples have Asthma and 23 Do not. I subset the phyloseq object into two groups: those with asthma and control. 
I used the following command to get the top taxa for the Asthma group and Control:

#Get most abundant taxa for asthma group
top10_A <- sort(taxa_sums(asthma), decreasing = TRUE)[1:10]
# get most abundant for control
top10_C <- sort(taxa_sums(no_asthma), decreasing = TRUE)[1:10]

> top10_A 
 ASV1  ASV2  ASV7  ASV8  ASV5  ASV9 ASV14 ASV17 ASV21 ASV16 
43807 17324  5872  5266  5009  4247  3637  3179  2771  2604 
> top10_C
 ASV1  ASV4  ASV2  ASV6 ASV10 ASV13 ASV11 ASV20  ASV5 ASV18 
15849  9569  9427  7496  4100  3747  3437  2818  2702  2474 

So the above are the top 10 most abundant ASVs. How do I access their taxonomy(GENUS) and is this the best way to find the most abundant GENUS specifically per control/asthma group?

Thank You.

**Comment by sarakn97** [comment]
I believe that I figured it out with tax_glom? 
```
# Merges ASVs that have the same taxonomy rank (Genus)
psga = tax_glom(asthma, taxrank = "Genus")

# Calculate taxa sum of the selected samples
ps.tA<- transform_sample_counts(psga, function(OTU) OTU/sum(OTU))

top10A_counts<- sort(taxa_sums(ps.tA), decreasing = TRUE)[1:10]
top10_A <- names(sort(taxa_sums(ps.tA), decreasing = TRUE))[1:10]
taxA <- tax_table(asthma)[top10_A,]
```

**Comment by gmteunisse** [comment]
Yes, `tax_glom` is the right function. However, you should set `NArm = F` to avoid discarding ASVs without a genus annotation - if not, your relative abundances will not be correct.
